src/components/AreaChart.js

```diff
@@ -63,72 +70,58 @@ const AreaChart = () => {
-      d.forEach((i) => {
-        i.date = parseDate(i.date);
-        i.price = Number(i.price);
-      });
-      setData(d);
-    });
-    return () => undefined;
-  }, []);
-
   return (
-    <div className="wrapper">
-      <div className="chartTitle">
-        AREACHART - USD to RUB Exchange Rates, 2020
-      </div>
-      <svg
-        viewBox={`-55 -55 ${width + margin.left + margin.right} 
-                            ${height + margin.top + margin.bottom}`}
-        onMouseMove={handleMouseMove}
-        onMouseLeave={handleMouseLeave}
-      >
-        <g className="axis" ref={getYAxis} />
-        <g
-          className="axis xAxis"
-          ref={getXAxis}
-          transform={`translate(0,${height})`}
-        />
-        <path fill={color} d={areaPath} opacity={0.3} />
-        <path strokeWidth={2} fill="none" stroke={color} d={linePath} />/
-        <text
-          transform={"rotate(-90)"}
-          x={0 - height / 2}
-          y={0 - margin.left}
-          dy="1em"
-          textAnchor="middle"
-        >
-          {"$1 USD equals"}
-        </text>
-        {data.map((item, index) => {
-          return (
-            <g key={index}>
-              <text
-                fill="#666"
-                x={getX(item.date)}
-                y={getY(item.price) - 20}
-                textAnchor="middle"
-              >
-                {index === activeIndex ? `₽ ${item.price}` : ""}
-              </text>
-              <circle
-                cx={getX(item.date)}
-                cy={getY(item.price)}
-                r={index === activeIndex ? 2 : 1}
-                fill={color}
-                strokeWidth={index === activeIndex ? 2 : 0}
-                stroke="#fff"
-                style={{ transition: "ease-out .1s" }}
-              />
-            </g>
-          );
-        })}
-      </svg>
-    </div>
+    <svg
+      viewBox={`-40 0 ${width + margin.left + margin.right} ${height}`}
+      onMouseMove={handleMouseMove}
+      onMouseLeave={handleMouseLeave}
+      fontFamily="Roboto Condensed, sans-serif"
+    >
+      <g className="axis" ref={getYAxis} strokeWidth={1} opacity={0.2} />
+      <g
+        className="axis xAxis"
+        ref={getXAxis}
+        transform={`translate(0,${height})`}
+        opacity={0.3}
+        fontVariant="all-small-caps"
+      />
+      <path fill={color} d={areaPath} opacity={0.3} />
+      <path strokeWidth={2} fill="none" stroke={color} d={linePath} />/
+      <text
+        transform={"rotate(-90)"}
+        x={0 - height / 2}
+        y={0 - margin.left}
+        dy="1em"
+        textAnchor="middle"
+      />
+      {data.map((item, index) => {
+        return (
+          <g key={index}>
+            <text
+              fill="#666"
+              x={getX(item.date)}
+              y={getY(item.price) - 20}
+              textAnchor="middle"
+            >
+              {index === activeIndex ? (
+                <tspan fontSize="large">₽ {item.price}</tspan>
```

Comment:
The AreaChart component could still be improved. This particular implementation requires data to be an array of objects with `price` and `date` properties. So if you try to reuse this component, you would also need to pass the data in the same format. Which makes it a little less intuitive if you try to plot a graph for something else.

Reply:
I'd use something like `x` and `y`. So depending on what data I want to show with this graph, all I need is to map the values to the respective axes. And it would make sense as to what value is plotted where.

Reply:
Yes Binal. Currently it is working for .csv format data. For json files it has to be created with different logic.

Reply:
> I'd use something like `x` and `y`. So depending on what data I want to show with this graph, all I need is to map the values to the respective axes. And it would make sense as to what value is plotted where.

Ok Binal. Got it